modules/aioseop_sitemap.php

```diff
@@ -1879,17 +1881,26 @@ public function get_sitemap_index_filenames() {
 
 			$files[] = array( 'loc' => aioseop_home_url( '/' . $prefix . '_addl' . $suffix ) );
 
-			if ( ! empty( $options[ "{$this->prefix}posttypes" ] ) ) {
+			$post_types = $options[ "{$this->prefix}posttypes" ];
+			if ( is_array( $aioseop_options['aiosp_cpostnoindex'] ) ) {
+				foreach ( $post_types as $index => $post_type ) {
+					if ( in_array( $post_type, $aioseop_options['aiosp_cpostnoindex'], true ) ) {
+						unset( $post_types[ $index ] );
```

Comment:
@EkoJR I don't think this will work for those posts that have been marked as NOINDEX in general settings but have been overridden to INDEX in the post settings. In this case, the sitemap index should feature the sitemap of that post type with the posts that have been overridden as INDEX. 

Also adding text comments will make this clearer, even below.

Reply:
I added a comment.

However, I've already addressed this. There is no INDEX on a NOINDEX setting.

https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/1382#issuecomment-477178336
https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/1382#issuecomment-478243111

Reply:
@EkoJR **currently** there is a INDEX option on a NOINDEX setting. I'm referring to the below in case there is any confusion.

![image](https://user-images.githubusercontent.com/12953439/56260519-25445500-60f5-11e9-9da3-40a022153975.png)

In https://github.com/semperfiwebdesign/all-in-one-seo-pack/issues/1382#issuecomment-477178336 you have proposed elimination of that.

In view of this, the current PR does not support INDEX option on a NOINDEX setting, which is what I've pointed out. Unless this is fixed or the option removed as per your suggestion, this continues to remain unfixed.

Reply:
@contactashish13 and @EkoJR To prevent this from bouncing back and forth between you, let's move this forward as is.  We know this doesn't solve the problem of overriding the default noindex setting for a post type on a single post.  Nobody has a solution for this and we know the code needs much more work in order to support this but we shouldn't hold  this PR up for that.  This PR solves the immediate problem where 1). a user can set a post type to noindex and it will be excluded from the sitemap, and 2). a user can set an individual post to noindex and it will be excluded from the sitemap.

Reply:
I still want to take another look at this. I eventually found how to trigger the setting, but I did investigate the code and didn't see any checks for it. So, it's kind of questionable.

Reply:
@wpsmort actually I did propose a solution for this in my earlier PR. That still holds true even if this PR supersedes that.

@EkoJR it would be great if you can copy the truth table from my PR and add as a code comment so that in the future, one does not have to go through the entire process again.